### PR TITLE
Torques and vels

### PR DESCRIPTION
**Pull request by 64-B1T**

### CODE REVIEW

File: robot_link.py [hunk 1]
```diff
@@ -18,6 +18,15 @@ def __init__(self, robot):
         self.joint_maxs = []
         self.methods_bound = [0, 0, 0, 0]
 
+        #Support for Collision Checking
+        self.link_names = None
+        self.vis_props = None #Used as a backup if col props unavailable
+        self.col_props = None #Populate first
+        #Col/Vis Props Format:
```
Comment (mcevoyandy) [inline robot_link.py]: remove or todo

**Review by mcevoyandy — COMMENTED** [review]
Difficult to review since also shows many of the same commits in #4. Change PR to merge into collision_detection, or finalize #4 and rebase this.

File: workspace_analyzer.py [hunk 2]
```diff
@@ -585,6 +585,65 @@ def parallel_process_point_cloud(self, num_poses, desired_poses, sphere, true_re
...
+        Args:
+            manipulability_space: results from a manipulability analysis
+            joint_goal_norm: goal joint norm
+
+        Results:
+            augmented results with joint vals added at each point
+        """
+        inds = [3, 6]
+        if angular:
+            inds = [0, 3]
```
Comment (mcevoyandy) [inline workspace_analyzer.py]: why [3, 6] and [0, 3]?

File: robot_link.py (hunk 1 above)
Comment (64-B1T) [reply robot_link.py]: Resolved; kept

File: workspace_command_line.py [hunk 3]
```diff
@@ -243,6 +257,27 @@ def save_to_file(self, results, out_file_name):
...
+        for r in results:
+            min = np.inf
+            for j in r:
+                if min > max(j[1]):
+                    min = max(j[1])
+            col = score_point(1 - min)
+            DrawRectangle(
+                tm([r[0][0], r[0][1], r[0][2], 0, 0, 0]),
+                [grid_rez]*3, ax, c=col, a=TRANSPARENCY_CONSTANT)
+            #disp('Plotting')
```
Comment (mcevoyandy) [inline workspace_command_line.py]:
```suggestion
```

**Review by mcevoyandy — APPROVED** [review]

File: workspace_analyzer.py (hunk 2 above)
Comment (64-B1T) [reply workspace_analyzer.py]: [0,3[ represents the angular component of a twist or wrench, and 3:6 represents the linear component . Added a comment explaining.

**Comment by 64-B1T** [comment]
+1

**Pull request merged**